src/core/highlight-vars.js

```diff
@@ -0,0 +1,102 @@
+
+/**
+ * Module core/highlight-vars
+ * Highlights occurrences of a <var> within a section on click.
+ * Set `conf.highlightVars = true` to enable.
+ * Removes highlights from <var> if clicked anywhere else.
+ * All is done while keeping in mind that exported html stays clean 
+ * on export.
+ */
+import { sub } from "core/pubsubhub";
+import hlVars from "deps/text!core/css/var.css";
+
+export const name = "core/highlight-vars";
+
+export function run(conf) {
+  if (!conf.highlightVars) {
+    return;
+  }
+  const styleElement = document.createElement("style");
+  styleElement.textContent = hlVars;
+  styleElement.classList.add("removeOnSave");
+  document.head.appendChild(styleElement);
+
+  document.addEventListener("click", initHighlight);
+
+  // remove highlights, cleanup empty class/style attributes
+  sub("beforesave", outputDoc => {
+    [...outputDoc.querySelectorAll("var.respec-active")].forEach(
+      removeHighlight
+    );
+  });
+}
+
+// availability of highlight colors.
+const HL_COLORS = new Map([
+  // [ background:String, available:Boolean ]
+  // or [ background,color:String, available:Boolean ]
+  ["yellow", true],
+  ["#726012,#fff", true],
+  ["#c78e00", true],
+  ["#3949ab,#fff", true],
+  ["#F720E5", true],
+]);
+
+function getHighlightColor(target) {
+  // return current colors if applicable
+  const bg = target.style.getPropertyValue("--respec-background-color");
+  const col = target.style.getPropertyValue("--respec-color");
+  if (bg) return `${bg}${col ? `,${col}` : ""}`;
+
+  // first color preference
+  if (HL_COLORS.get("yellow") === true) return "yellow";
+
+  // otherwise get some other available color
+  return [...HL_COLORS.keys()].find(c => HL_COLORS.get(c) === true) || "yellow";
+}
+
+function highlightVars(target) {
+  const textContent = target.textContent.trim();
+  const parent = target.closest("section");
+  const highlightColor = getHighlightColor(target);
+
+  const varsToHighlight = [...parent.querySelectorAll("var")].filter(
+    el => el.textContent.trim() === textContent
+  );
+
+  // update availability of highlight color
+  const colorStatus = varsToHighlight[0].classList.contains("respec-active");
+  HL_COLORS.set(highlightColor, colorStatus);
+
+  // highlight vars
+  const [background, color] = highlightColor.split(",");
```

Comment:
I'm wondering if we should put HL_COLOR's into js/core/css/var.css, then we can name them:

```JS
const HL_COLORS = new Map([
  ["respec-hl-yellow", true],
  ["respec-hl-brown", true],
]);
```

And so on... then we don't need to deal foreground/background colors here. We just use the next CSS class available in the map.

Reply:
Was looking for nice contrasting color palettes: quite like http://colorpalettes.net/color-palette-3694/